app/forms/DateOfDeathFormProvider.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import java.time.LocalDate
+
+import forms.mappings.Mappings
+import javax.inject.Inject
+import play.api.data.Form
+
+class DateOfDeathFormProvider @Inject() extends Mappings {
+
+  def withConfig(trustStartDate: LocalDate, prefix: String): Form[LocalDate] =
+    Form(
+      "value" -> localDate(
+        invalidKey     = s"$prefix.error.invalid",
+        allRequiredKey = s"$prefix.error.required.all",
+        twoRequiredKey = s"$prefix.error.required.two",
+        requiredKey    = s"$prefix.error.required"
+      ).verifying(firstError(
+        maxDate(trustStartDate, s"$prefix.error.future", "day", "month", "year"),
+        minDate(LocalDate.of(1500,1,1), s"$prefix.error.past", "day", "month", "year")
```

Comment:
Should the min date not be their date of birth? Or are we not guaranteed to have that?

Reply:
Good question I'll ask Wendy